app/routes/tasks.py

```diff
@@ -0,0 +1,202 @@
+
+        for task in tasks:
+            task_response.append(task.to_dict())     
+        return  make_response(jsonify(task_response),200)
+
+    except Exception:
+        abort(400)
+
+
+''' GET <task_id>  - this function returns a task when 
+                    an unique id is passed in the request body
+                    if not found returns 404 error
+'''
+@task_bp.route("/<task_id>", methods=["GET"])
+def read_one_drink(task_id):
+    task = get_task_by_id(task_id)
+
+    try:
+        if task.goal_id:
+            task_response = {"task": task.to_dict_goal_task() }                     
+        else:
+            task_response = {"task": task.to_dict() }
+
+        return jsonify(task_response), 200
+
+    except Exception: 
+        abort(400)
+
+
+''' DELETE      - this function handles the deletion of a task by its id
+'''
+
+@task_bp.route("/<task_id>", methods=['DELETE'])
+def delete_one_task(task_id):
+    task = get_task_by_id(task_id)
+    try:
+        db.session.delete(task)
+        db.session.commit()
+        response_body = {"details": f'Task {task.task_id} "{task.title}" successfully deleted'}
+
+        return make_response(response_body, 200)
+
+    except Exception: 
+        abort(400)
+
+
+
+''' PUT   - this function updates a task by its id'''
+
+@task_bp.route("/<task_id>", methods=["PUT"])
+def update_task(task_id):
+    task = get_task_by_id(task_id)
+
+    try:
+        request_body = request.get_json()
+        if not request_body: 
+            abort(400)
+
+        if "title" in request_body:
+            task.title = request_body["title"]
+        if "description" in request_body:
+            task.description = request_body["description"]
+    
+        # commit session after patching 
+        db.session.commit()
+
+        response_body = {
+            "task": task.to_dict()
+        }
+        return make_response(response_body, 200)
+
+    except Exception: 
+        abort(400)
+
+
+# use a dynamic route to handle is complete and else - CUSTOM ROUTE
+# SLACK API
+@task_bp.route("/<task_id>/mark_complete", methods=["PATCH"])
+def mark_complete(task_id):
+    task =  get_task_by_id(task_id)
+    date = datetime.utcnow()
+
+    try:
+        if task.completed_at == None:
+            task.completed_at = date
+        db.session.commit()
+        response_body = {
+                "task": task.to_dict()} 
+
+        # send slack notificatios
+        message = f"Someone just completed the task {task.title}"     
+        post_msg_slack(message)
+
+        return response_body
+
+    except Exception:
+        abort(400)
+
+
+@task_bp.route("/<task_id>/mark_incomplete", methods=["PATCH"])
+def mark_incomplete(task_id):
+    task =  get_task_by_id(task_id)
+
+    try:
+        task.completed_at = None
+        db.session.commit()
+    
+        # send slack notification 
+        message = f"Someone just completed the task {task.title}"     
+        post_msg_slack(message)
```

Comment:
If you can't post to Slack this returns a 400 (even though the request is fine).